src/libraries/System.Data.OleDb/src/OleDbConnection.COMWrappers.cs

```diff
@@ -0,0 +1,104 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.ComponentModel;
+using System.Data.Common;
+using System.Data.ProviderBase;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace System.Data.OleDb
+{
+    public sealed partial class OleDbConnection
+    {
+        internal static unsafe Exception? ProcessResults(OleDbHResult hresult, OleDbConnection? connection, object? src)
+        {
+            if ((0 <= (int)hresult) && ((null == connection) || (null == connection.Events[EventInfoMessage])))
+            {
+                SafeNativeMethods.Wrapper.ClearErrorInfo();
+                return null;
+            }
+
+            // ErrorInfo object is to be checked regardless the hresult returned by the function called
+            Exception? e = null;
+            IntPtr pErrorInfo;
+            OleDbHResult hr = UnsafeNativeMethods.GetErrorInfo(0, &pErrorInfo);  // 0 - IErrorInfo exists, 1 - no IErrorInfo
+            if ((OleDbHResult.S_OK == hr) && (IntPtr.Zero != pErrorInfo))
+            {
+                UnsafeNativeMethods.IErrorInfo errorInfo = (UnsafeNativeMethods.IErrorInfo)OleDbComWrappers.Instance
```

Comment:
Can we just split this line out into separate partial methods between the two? That way we don't need to duplicate this whole method?